src/renderer/components/CustomRouter/redirects.ts

```diff
@@ -0,0 +1,22 @@
+import { useSelector } from 'react-redux';
+import { CurrentView } from '../../../shared/redux/slices/currentViewSlice';
+import { activePublication } from '../../../shared/redux/slices/loadPublicationsSlice';
+import { VIEWS } from '../../constants/Views';
+
+const useViewRedirects = (currentView: CurrentView): CurrentView => {
+  let redirectedView = currentView;
+
+  const isProjectActive = Boolean(useSelector(activePublication));
+  if (
+    [VIEWS.FILES, VIEWS.CHANGES, VIEWS.SETTINGS, VIEWS.PROJECT].includes(
+      currentView.view
+    ) &&
+    !isProjectActive
+  ) {
+    redirectedView = { ...currentView, ...{ view: VIEWS.NO_ACTIVE_PROJECT } };
+  }
+
+  return redirectedView;
+};
```

Comment:
Yup. Also removed the same mistake repeated in tests.

Reply:
I've resolved this conversation prematurely and forgot to implement the rest of your suggestions 😆 

I have omitted the early return - I've written it this way, because it allows us to extend it later with multiple conditions if needed. Unless you would rather have it solved in a less procedural way?

Reply:
@LostInStatic I'll just move the enum destructuring outside the hook and merge this MR. 🚀